src/web_interface/components/database_routes.py

```diff
@@ -152,18 +151,17 @@ def _build_search_query(self):
             self._add_hash_query_to_query(query, request.form['hash_value'])
         if 'tags' in request.form and request.form['tags']:
             self._add_tag_query(query)
-        print(query)
         return json.dumps(query)
 
     @staticmethod
     def _add_multiple_choice(query, key):
-        new_query = [{key: name} for name in dict(request.form.lists())[key]]
-        query.setdefault('$or', []).extend(new_query)
+        for name in dict(request.form.lists())[key]:
+            query[key] = {"$in": [name]}
 
     @staticmethod
     def _add_tag_query(query):
-        tag_query = [{f'tags.{tag}': 'secondary' for tag in dict(request.form.lists())['tags']}]
-        query.setdefault('$or', []).extend(tag_query)
+        for tag in dict(request.form.lists())['tags']:
+            query[f'tags.{tag}'] = {"$in": ['secondary']}
```

Comment:
The "$in" query doesn't really work here because the values ("secondary") are the same but the keys are different. If should work if we simply leave it as it was if it is the only "$or" query, though.